pkg/importer/http-datasource.go

```diff
@@ -115,17 +118,25 @@ func (hs *HTTPDataSource) Info() (ProcessingPhase, error) {
 		klog.Errorf("Error creating readers: %v", err)
 		return ProcessingPhaseError, err
 	}
-	// The readers now contain all the information needed to determine if we can stream directly or if we need scratch space to download
-	// the file to, before converting.
-	if !hs.readers.Archived && !hs.customCA && !hs.brokenForQemuImg && hs.readers.Convert {
-		// We can pass straight to conversion from the endpoint. No scratch required.
-		hs.url = hs.endpoint
+	if hs.brokenForQemuImg {
+		return ProcessingPhaseTransferScratch, nil
+	}
+	hs.url = hs.endpoint
+	if !hs.readers.Archived && hs.customCA == "" && hs.readers.Convert {
+		// We can pass straight to conversion from the endpoint
 		return ProcessingPhaseConvert, nil
 	}
-	if !hs.readers.Convert {
-		return ProcessingPhaseTransferDataFile, nil
+	hs.n = image.NewNbdkitCurl("/var/run/nbdkit.pid", hs.customCA)
+	if hs.readers.ArchiveGz {
+		hs.n.AddFilter(image.NbdkitGzipFilter)
+		klog.V(2).Infof("Added nbdkit gzip filter")
+	}
+	if hs.readers.ArchiveXz {
+		hs.n.AddFilter(image.NbdkitXzFilter)
+		klog.V(2).Infof("Added nbdkit xz filter")
 	}
-	return ProcessingPhaseTransferScratch, nil
+
+	return ProcessingPhaseConvertNbd, nil
```

Comment:
I think you can do something like this here:
```go
qemuOperations = image.NewNbdkitOperations(hs.GetNbdkit())
return ProcessingPhaseConvert, nil
```
And have the same result as you have now without all the extra stuff in data processor.